### PR TITLE
Update default value for PageSize

### PR DESCRIPTION
**Pull request by jsharmbh**
### Background:- 
- There is a bug [PTL-5638](https://jciproducts.atlassian.net/browse/PTL-5638) in the PVT application, where PVT is taking a very long time to get the MUI data for a huge Metasys Site.
for example, Getting all Equipment calls takes approx 40-50 mins (where equipment count is approx 8500).

This PR contains a fix, where increasing the PageSize to 1000 (from 100) has improved the performance. for example, now getting Equipment data from the same call takes approx. 15 mins.

### Help Needed:- 
- Could increasing the PageSize to 1000 impact any other API calls?
- Any suggestion for other ways to increase the performance of the API calls?
- There is another method in BasicServiceProvider named [ToMetasysObject](https://github.com/metasys-server/basic-services-dotnet/blob/015d41d94e8e8c3ba133a2e88d9d9b178d953af3/MetasysServices/BasicServiceProvider.cs#L130-L139) which also takes times to parse the JSON response into MetasysObject. I am not sure if that method could also be improved further to increase performance. Could you please take a look at this method as well?

@Federico-Artioli @michaelgwelch

### CODE REVIEW

**Comment by michaelgwelch** [comment]
Looks good to me, but I'll suggest that @Federico-Artioli do the actual merge.

**Comment by michaelgwelch** [comment]
> Could increasing the PageSize to 1000 impact any other API calls?

It should be okay. I wouldn't run other API jobs at the same time. And don't make multiple calls for equipment in parallel.

File: MetasysServices/BasicServiceProvider.cs
Comment (michaelgwelch) [inline MetasysServices/BasicServiceProvider.cs:563]: @Federico-Artioli you may want to consider exposing `pageSize` as a parameter with a default value of 100 (or 1000) in the `GetAllAvailablePagesAsync` method.

**Comment by michaelgwelch** [comment]
@jsharmbh I just tested this api with our internal server and it took 1 minute and 25 seconds to fetch 100 equipment. I now seem to recall that we have a defect open against this api. I'm going to check.

File: MetasysServices/BasicServiceProvider.cs
Comment (jsharmbh) [reply MetasysServices/BasicServiceProvider.cs:563]: Yes, `GetAllAvailablePagesAsync` method is already having a parameter that is of type dictionary and the purpose of the dictionary is to expose properties for example `page`, `pagesize`, etc.

In my opinion, the public methods that the MetasysServices library is exposing for example [Get all equipments](https://github.com/metasys-server/basic-services-dotnet/blob/a03b5afa5c8e4eac26569a59b624cac06fe849aa/MetasysServices/Equipments/EquipmentServiceProvider.cs#L51-L54), etc. needs to expose such properties, so that the consumer can get the data based on their provided `pagesize` values.

**Comment by jsharmbh** [comment]
> @jsharmbh I just tested this api with our internal server and it took 1 minute and 25 seconds to fetch 100 equipment. I now seem to recall that we have a defect open against this api. I'm going to check.

there is a generic method [ToMetasysObject](https://github.com/metasys-server/basic-services-dotnet/blob/015d41d94e8e8c3ba133a2e88d9d9b178d953af3/MetasysServices/BasicServiceProvider.cs#L130-L139) which gets called recursively to find the asked object and it also takes time to complete depending on the depth of the JSON response. We can also take a look at this method _if this can be optimized_.

File: MetasysServices/BasicServiceProvider.cs
Comment (Federico-Artioli) [reply MetasysServices/BasicServiceProvider.cs:563]: Yes I also agree that exposing a parameter as `pageSize` in some public methods as `Get All Equipment` will help the user to set the required value when they call those methods.

**Review by Federico-Artioli — APPROVED** [review]

**Comment by michaelgwelch** [comment]
> **_Update:-_** Ohh.. you tested it using REST API call, then [ToMetasysObject](https://github.com/metasys-server/basic-services-dotnet/blob/015d41d94e8e8c3ba133a2e88d9d9b178d953af3/MetasysServices/BasicServiceProvider.cs#L130-L139) is not applicable in your testing. The time you mention is a completely because of the REST API response.

Yep, that is correct @jsharmbh . I just called it directly from the command line using [MetasysRestClient](https://www.powershellgallery.com/packages/MetasysRestClient/) for powershell

```pwsh
~> Measure-Command { imm /equipment > equipment.json }
Days              : 0
Hours             : 0
Minutes           : 1
Seconds           : 25
Milliseconds      : 318
Ticks             : 853188758
TotalDays         : 0.000987486988425926
TotalHours        : 0.0236996877222222
TotalMinutes      : 1.42198126333333
TotalSeconds      : 85.3188758
TotalMilliseconds : 85318.8758
```

**Comment by jsharmbh** [comment]
@michaelgwelch, could you approve this PR? PR needs at least 2 approvals for merging.

@Federico-Artioli, Once the PR gets merged, will the NuGet package will be generated automatically or it requires some additional manual intervention?
We need the updated NuGet package to consume in the PVT application.

**Comment by Federico-Artioli** [comment]
@jsharmbh Yes, after I will merge the PR I will have to build the new NuGet package (at the moment it is still required some manual actions)

**Pull request merged**
